File: Chapter_3/TemperatureConverter/C/main.c
#include <stdio.h>

// Convert fahrenheit to celsius
float fahrenheit_to_celsius(int fahrenheit) {
    return (fahrenheit - 32) / 1.8;
}

int main() {
    for (int fahrenheit = -40; fahrenheit < 41; fahrenheit++) {
        float celsius = fahrenheit_to_celsius(fahrenheit);
        printf("%dF is %fC\n", fahrenheit, celsius);
    }
}

File: Chapter_3/TemperatureConverter/Rust/src/main.rs
// Convert temperatures from fahrenheit to celsius
fn fahrenheit_to_celsius(fahrenheit_temperature: i32) -> f32 {
    // Formula is (F - 32) / 1.8
    let temp_less_32 = fahrenheit_temperature - 32;
    return (temp_less_32 as f32) / 1.8;
}

fn main() {
    for fahrenheit in -40..41 {
        let celsius = fahrenheit_to_celsius(fahrenheit);
        println!("{}°F is {}°C", fahrenheit, celsius);
    }
}
